pkg/vulnerabilityreport/builder.go

```diff
@@ -143,6 +143,7 @@ type ReportBuilder struct {
 	container  string
 	hash       string
 	data       v1alpha1.VulnerabilityReportData
+	reportTTL  string
```

Comment:
Can we use *time.Duration instead of string here? To fail fast we should not pass invalid duration literals to the builder.

Reply:
We could do that and manage the conversion from time.Duration to string inside the function. We need to do this thanks to that a annotation always is `map[string]string`.

If we don't want to do that we could also leave the idea of using a annotation and instead go over to using the status field to store the ttl data.

We could do it in a similar way that certmanager have done it in there status field.

```
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: something
spec:
  commonName: something
  duration: 8760h0m0s
  issuerRef:
    name: intermediate-ca
  privateKey:
    algorithm: ECDSA
    size: 521
  revisionHistoryLimit: 1
  secretName: something
  usages:
  - server auth
  - client auth
status:
  conditions:
  - lastTransitionTime: "2021-06-21T10:42:17Z"
    message: Certificate is up to date and has not expired
    observedGeneration: 3
    reason: Ready
    status: "True"
    type: Ready
  notAfter: "2022-07-26T08:51:28Z"
  notBefore: "2021-07-26T08:51:28Z"
  renewalTime: "2022-03-26T16:51:28Z"
  revision: 2
```

That way we could also say the estimated time when the report will be deleted.
The "bad" thing is that users won't be able to change the TTL on the report, but I don't think I see this as a bad thing overall.
Another thing with the status field is that we could also show the TTL value when performing a get on the vulnerabilityreport and show another field. Don't know if we can do that with a feature flag or not though.

```shell
kubectl get vulnerabilityreports.aquasecurity.github.io
NAME                  REPOSITORY                       TAG       SCANNER   AGE          TTL
pod1                       thanos/thano                       v0.24.0      Trivy          14d            24h
```

This should probably be discussed in the design document. I will link back to this comment to make it visible.

Reply:
VulnerabilityReport (and other reports managed by Starboard) are a bit different than Certificate and similar kinds. Mainly because we use them as data holder. Therefore the root property of the VulnerabilityReport is Report (I wish I have called it Data) and we do not have Spec. According to K8s API machinery conventions you either use Spec+Status or a single property with an arbitrary name (e.g. Report or Data).

Reply:
Annotations are good to evaluate ideas without introducing new properties. Once we add a property it's hard to deprecate it. So I'd suggest to merge this PR with the annotation. We could always improve it if it's not enough.

For a helper method and parsing the annotation we could try defining it on the VulnerabilityReport type. Then whether we read it from annotation (today) or property (in few weeks or months) will be an implementation detail.

Reply:
Anyway, I think in this particular line we inject TTL of type time.Duration, which is parsed by the env config module. If the value was incorrect the operator would not even start.

Reply:
Ahh didn't know that you couldn't use status if you used your own custom type. Thanks for the explanation